src/st/no0/e_skelerang.c

```diff
@@ -2,17 +2,335 @@
+        if (self->animCurFrame == 30) {
+            PlaySfxPositional(SFX_THROW_WEAPON_SWISHES);
+            entityTwo = &PLAYER;
+            entity = self + 1;
+            entity->posX.i.hi = entityTwo->posX.i.hi;
+            entity->posY.i.hi = entityTwo->posY.i.hi;
+
+            entity = self + 2;
+            entity->posX.i.hi = self->posX.i.hi;
+            entity->posY.i.hi = self->posY.i.hi;
+            entity->animCurFrame = 43;
+            entity->step++;
+
+            entity = self + 3;
+            entity->posX.i.hi = self->posX.i.hi;
+            entity->posY.i.hi = self->posY.i.hi;
+            entity->animCurFrame = 43;
+            entity->step++;
+        }
+        break;
+    case 6:
+        entity = self + 2;
+        entityTwo = self + 3;
+        if (entity->step == 3 && entityTwo->step == 3) {
+            entity->step = 1;
+            entityTwo->step = 1;
+            SetStep(7);
+        }
+        break;
+    case 7:
+        if (!AnimateEntity(D_us_80181F34, self)) {
+            self->ext.skelerang.unk84++;
+            if (self->ext.skelerang.unk84 > 2) {
+                SetStep(9);
+            } else {
+                SetStep(4);
+            }
+        }
+        break;
+    case 8:
+        switch (self->step_s) {
+        case 0:
+            if (!AnimateEntity(D_us_80181F38, self)) {
+                self->animFrameIdx = 0;
+                self->animFrameDuration = 0;
+                self->step_s++;
+            }
+            break;
+        case 1:
+            AnimateEntity(D_us_80181F60, self);
+            if ((GetDistanceToPlayerX() > 80) ||
+                (GetDistanceToPlayerY() > 48)) {
+                self->animFrameIdx = 0;
+                self->animFrameDuration = 0;
+                self->step_s++;
+            }
+            break;
+        case 2:
+            if (!AnimateEntity(D_us_80181F4C, self)) {
+                SetStep(9);
+            }
+            break;
+        }
+        break;
+    case 10:
+        entity = AllocEntity(&g_Entities[224], &g_Entities[256]);
+        if (entity != NULL) {
+            CreateEntityFromEntity(2, self, entity);
+            entity->params = 2;
+            entity->posY.i.hi += 24;
+        }
+        DestroyEntity(self);
+        break;
+    case 12:
+        for (i = 0; i < 8; i++) {
+            entity = AllocEntity(&g_Entities[224], &g_Entities[256]);
+            if (entity != NULL) {
+                MakeEntityFromId(E_SKELERANG, self, entity);
+                entity->flags = FLAG_DESTROY_IF_OUT_OF_CAMERA |
+                                FLAG_POS_CAMERA_LOCKED | FLAG_UNK_2000;
+                entity->hitboxState = 0;
+                entity->animCurFrame = i + 44;
+                entity->params = i;
+                entity->facingLeft = self->facingLeft;
+                if ((GetSideToPlayer() & 1) ^ 1) {
+                    // Sus store to Entity 0xA
+                    F(entity->velocityX).i.hi = -(Random() & 3);
```

Comment:
A suggestion from @bismurphy to pull this into an f32 macro - I guess it's possible `velocityX` / `velocityY` are actually f32s and accessed this way everywhere but there's no other evidence of that in the codebase besides here so I've just macroed for now.